core/src/impl/KokkosExp_IterateTileGPU.hpp

```diff
@@ -471,14 +310,13 @@ struct DeviceIterateTile<4, PolicyType, Functor, void> {
 
   KOKKOS_IMPL_DEVICE_FUNCTION
   void exec_range() const {
-    // LL
     if (PolicyType::inner_direction == PolicyType::Left) {
       const index_type temp0  = m_policy.m_tile_end[0];
       const index_type temp1  = m_policy.m_tile_end[1];
       const index_type numbl0 = (temp0 <= max_blocks ? temp0 : max_blocks);
       const index_type numbl1 =
           (temp0 * temp1 > max_blocks
-               ? index_type(max_blocks / numbl0)
+               ? static_cast<index_type>(max_blocks / numbl0)
```

Comment:
Why is a C++-style cast insufficient here? If it were to be a non-numeric type, it's entirely possible that the syntax implying a constructor is reasonable, and I've never had any trouble with thinking of integers as copy constructible. (It works fine either way; I'm just curious about why you don't like C++-style casts).

Reply:
Mainly for conformity. We were using the `static_cast` syntax a lot more. 
Regardless, they look pretty close to C-style casts and it's maybe a personal preference to use `static_cast` rather than constructor calls for numeric types.